nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js

```diff
@@ -470,6 +470,214 @@ nf.UsersTable = (function () {
+            policyLabel += 'process group ';
+        } else if (resource.startsWith('/remote-process-groups')) {
+            policyLabel += 'remote process group ';
+        } else if (resource.startsWith('/reporting-tasks')) {
+            policyLabel += 'reporting task ';
+        } else if (resource.startsWith('/templates')) {
+            policyLabel += 'template ';
+        }
+
+        if (dataContext.component.componentReference.permissions.canRead === true) {
+            policyLabel += dataContext.component.componentReference.component.name;
+        } else {
+            policyLabel += '<span class="unset">' + dataContext.component.componentReference.id + '</span>'
+        }
+
+        return policyLabel;
+    };
+
+    /**
+     * Initializes the user policies table.
+     */
+    var initUserPoliciesTable = function () {
+
+        // function for formatting the human readable name of the policy
+        var policyDisplayNameFormatter = function (row, cell, value, columnDef, dataContext) {
+            // if the user has permission to the policy
+            if (dataContext.permissions.canRead === true) {
+                // check if Global policy
+                if (nf.Common.isUndefinedOrNull(dataContext.component.componentReference)) {
+                    return globalResourceParser(dataContext);
+                }
+                // not a global policy... check if user has access to the component reference
+                return componentResourceParser(dataContext);
+            } else {
+                return '<span class="unset">' + dataContext.id + '</span>';
+            }
+        };
+
+        // function for formatting the actions column
+        var actionsFormatter = function (row, cell, value, columnDef, dataContext) {
+            var markup = '';
+
+            if (dataContext.permissions.canRead === true) {
+                if (nf.Common.isDefinedAndNotNull(dataContext.component.componentReference)) {
+                    markup += '<div title="Go To" class="pointer go-to-user-policies fa fa-long-arrow-right" style="float: left;"></div>';
+                }
+            }
+
+            return markup;
+        };
+
+        // function for formatting the action column
+        var actionFormatter = function (row, cell, value, columnDef, dataContext) {
+            var markup = '';
+
+            if (dataContext.permissions.canRead === true) {
+                markup += dataContext.component.action;
+            }
+
+            return markup;
+        };
+
+        var userPoliciesColumns = [
+            {id: 'policy', name: 'Policy', sortable: true, resizable: true, formatter: policyDisplayNameFormatter, width: 150},
+            {id: 'action', name: 'Action', sortable: true, resizable: false, formatter: actionFormatter, width: 50}
+        ];
+
+        // add the actions if we're in the shell
+        if ((top !== window) && nf.Common.canAccessPolicies()) {
```

Comment:
Policies will have their own specific permissions. No need to check if the user can access all policies here.